bitcoin/src/blockdata/transaction.rs

```diff
@@ -993,6 +993,51 @@ impl Transaction {
         // `Transaction` docs for full explanation).
         self.input.is_empty()
     }
+
+    /// Returns a reference to the input at `input_index` if it exists.
+    #[inline]
+    pub fn txin(&self, input_index: usize) -> Result<&TxIn, IndexOutOfBoundsError> {
```

Comment:
In 440dee61ad52cbcf0818d0d1a862abb585827c48:

Should be `tx_in` (and `tx_out`) to match capitalization of `TxIn`/`TxOut`. (Ok to do this in a separate commit in case you don't want to rebase your whole commit stack.)

Reply:
Rebasing is part of my job ;) This is tongue in cheek but I don't understand what is the difficulty with rebasing, I rebase my patch sets constantly all day and move changes up and down the patch set into the right place. I would also rebase on master constantly but I resist so that reviewers can do `git range-diff` and get less cluttered output.

Reply:
But yes, `tx_in` and `tx_out` are better names. Sorry I'm a bit prickly this week aren't I.

Reply:
All good :)

And usually rebases aren't too big a concern, but renaming a constant that's used in a bunch of future patches can be pretty annoying and lead to tons of conflicts.

Reply:
Well, you may be amused to hear that after being so facetious I had to rebase this PR about 10 times to get everything in the right place :)